lib/alexa_generator/interaction_model.rb

```diff
@@ -45,29 +45,39 @@ def initialize(intents, utterance_templates, slot_bindings)
     end
 
     def intent_schema
-      {
-          intents: @intents.values.map do |intent|
-            {
-                intent: intent.name,
-                slots: intent.slots.map do |slot|
-                  {
-                      name: slot.name,
-                      type: slot.type
-                  }
-                end
-            }
-          end
-      }
+      out = { intents: [] }
+
+      @intents.values.each do |intent|
+        hash = { intent: intent.name }
+        slots = intent.slots.map do |slot|
+          {
+              name: slot.name,
+              type: slot.type
+          }
+        end
+        
+        if slots.size > 0 || !intent.name =~ /^AMAZON/
```

Comment:
all the samples just had the string. and no slots. I thought it was nice to keep it tight.